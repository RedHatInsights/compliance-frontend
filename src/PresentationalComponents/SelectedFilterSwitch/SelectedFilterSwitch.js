import React from 'react';
import propTypes from 'prop-types';
import { Switch } from '@patternfly/react-core';

const SelectedFilterSwitch = ({ isChecked, setActiveFilter, ...props }) => (
  <Switch
    label="Selected only"
    isChecked={isChecked}
    onChange={() => setActiveFilter('selected', !isChecked)}
    {...props}
  />
);

SelectedFilterSwitch.propTypes = {
  isChecked: propTypes.bool,
  setActiveFilter: propTypes.func,
};

export default SelectedFilterSwitch;
