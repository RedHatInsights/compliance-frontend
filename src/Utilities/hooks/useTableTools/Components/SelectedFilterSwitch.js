import React from 'react';
import { Switch } from '@patternfly/react-core';
import propTypes from 'prop-types';

const SelectedFilterSwitch = ({ isChecked, setActiveFilter }) => (
  <Switch
    label="Selected only"
    isChecked={isChecked}
    onChange={() => setActiveFilter('selected', !isChecked)}
  />
);

SelectedFilterSwitch.propTypes = {
  isChecked: propTypes.bool,
  setActiveFilter: propTypes.func,
};

export default SelectedFilterSwitch;
