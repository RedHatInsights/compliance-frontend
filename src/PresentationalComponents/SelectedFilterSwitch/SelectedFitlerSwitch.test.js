import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SelectedFilterSwitch from './SelectedFilterSwitch';

describe('SelectedFilterSwitch', () => {
  it('expect to render without error', () => {
    const component = <SelectedFilterSwitch />;

    expect(toJson(shallow(component))).toMatchSnapshot();
  });

  it('expect to render without error', () => {
    const component = <SelectedFilterSwitch isChecked={false} />;

    expect(toJson(shallow(component))).toMatchSnapshot();
  });
});
