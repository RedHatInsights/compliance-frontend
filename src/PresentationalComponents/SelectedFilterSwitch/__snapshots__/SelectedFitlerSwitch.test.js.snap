// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFilterSwitch expect to render without error 1`] = `
<Switch
  aria-label=""
  isChecked={true}
  isDisabled={false}
  isReversed={false}
  label="Selected only"
  onChange={[Function]}
/>
`;

exports[`SelectedFilterSwitch expect to render without error 2`] = `
<Switch
  aria-label=""
  isChecked={false}
  isDisabled={false}
  isReversed={false}
  label="Selected only"
  onChange={[Function]}
/>
`;
